### ISSUE TITLE
contour widget has no pin button to integrate in canvas

### ISSUE THREAD

**Issue by jott3077** [CARTAvis/carta-frontend#1219]
The contour widget lacks a pin button in the top window to pin it onto the canvas. It should have the same docking options as any other widget.

**Comment by kswang1029** [comment]
It is intended to have the contour config as a "dialogue" not widget. Is there any particular use case that really requires the dialogue as a dockable widget? @arusstaylor @low-sky @Jordatious @cfee25 @veggiesaurus  thoughts?

**Comment by veggiesaurus** [comment]
I seem to remember @Jordatious asking for it to be treated as a widget as well. The question then is: do we limit it to a single widget, or can there be multiple widgets? Do we need a category of "single use" widgets?, such as the contour and Stokes?

**Comment by Jordatious** [comment]
Yes I agree it would be good to be able to pin, particularly since allowing this doesn't take away from it being able to float like currently. It also may deal with #932. Having "single use" widgets sounds sensible, although I could imagine that for the Stokes widget, a user might want several open with different image and region selections. Perhaps similarly for the contour widget and the "data source", although I'd have to go and check how this is used again.

**Comment by jott3077** [comment]
One disadvantge of the non-docking is that the floating widget eclipses the canvas behind. So when I wanted to do get the contours, I had to move it away to have access to the render config, animator etc. It would have been a lot easier to have it docked. 

It should be possible to have a few contour widgets at the same time for different images. That allows one to optimize the overlays. Good question what happens if multiple widgets operate on the same image. Maybe this would allow a second set of contours on the same image? 

If possible I would avoid single use widgets. I think most should be dockable. That includes also the file browser and even possibly the help pages.

**Comment by low-sky** [comment]
I think contours are a bit like a render configuration panel and is something that will be adjusted through the course of interactive visualization.  Thus, having a dockable widget rather than a dialogue would work a bit better for my use cases.  Like others, I usually had to drag aside the dialogue, and adjust parameters iteratively.
